sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -469,9 +469,16 @@ object StructType extends AbstractDataType {
           case leftField @ StructField(leftName, leftType, leftNullable, _) =>
             rightMapped.get(leftName)
               .map { case rightField @ StructField(_, rightType, rightNullable, _) =>
-                leftField.copy(
-                  dataType = merge(leftType, rightType),
-                  nullable = leftNullable || rightNullable)
+                Try {
```

Comment:
Could we use the JAVA style `try` and `catch`? See the https://github.com/databricks/scala-style-guide#exception-handling-try-vs-try